EmailSender/Model/TemplateRepository.php

```diff
@@ -0,0 +1,52 @@
+<?php
+declare(strict_types=1);
+
+namespace RLTSquare\EmailSender\Model;
+
+use Magento\Email\Model\ResourceModel\Template as TemplateResourceModel;
+use Magento\Email\Model\TemplateFactory;
+use Magento\Framework\Exception\NoSuchEntityException;
+use RLTSquare\EmailSender\Api\TemplateRepositoryInterface;
+
+/**
+ * @TemplateRepository
+ */
+class TemplateRepository implements TemplateRepositoryInterface
+{
+    /**
+     * @var TemplateFactory
+     */
+    protected TemplateFactory $templateFactory;
+
+    /**
+     * @var TemplateResourceModel
+     */
+    protected TemplateResourceModel $templateResourceModel;
+
+    /**
+     * @param TemplateFactory $templateFactory
+     * @param TemplateResourceModel $templateResourceModel
+     */
+    public function __construct(
+        TemplateFactory $templateFactory,
+        TemplateResourceModel $templateResourceModel
+    ) {
+        $this->templateFactory        = $templateFactory;
+        $this->templateResourceModel  = $templateResourceModel;
+    }
+
+    /**
+     * @param string $id
+     * @return int|null
+     * @throws NoSuchEntityException
+     */
+    public function getById(string $id)
+    {
+        $object = $this->templateFactory->create();
```

Comment:
Overcomplicated code - overcomplicated and it actually misses the approach, which is system.xml + config.xml + reading value for email template from config, see https://drive.google.com/file/d/1ed3KeOMXJoXriWg7CJeEh4rEGzFkU1vk/view?usp=share_link

Reply:
Thanks for the valuable feedback. I'll dive into this and do recommended rectifications and modification. I tried to update the email template on every update without loading every template in separate configuration. But unfortunately i won't be able to do that. I was thinking to automate approach with lesser and optimized code